Packs/RadwareCloudDDoSProtectionServices/Integrations/RadwareCloudDDoSProtectionServices/RadwareCloudDDoSProtectionServices.py

```diff
@@ -102,86 +104,141 @@ def add_fields(events: list[dict], evnet_type: str | None):
         event['source_log_type'] = evnet_type
 
 
-def fetch_events(client, last_run):
-    end_time = int(time.time() * 1000)
-    start_time = last_run.get('last_fetch_events', end_time-2)+1
-    continue_fetch_events = last_run.get('continue_fetch_events', None)
-    skip = 0
-    if continue_fetch_events:
-        end_time = continue_fetch_events.get('end_time')
-        start_time = continue_fetch_events.get('start_time')
-        skip = continue_fetch_events.get('fetched_events')
+def filter_documents(documents: List[Dict[str, Union[str, Dict]]],
+                     timestamp: Union[str, None], ids: List[str], data_type: str) -> List[Dict[str, Union[str, Dict]]]:
+    """
+    Filters out documents from the given list based on the specified timestamp and ids.
+    Args:
+        documents (List[Dict[str, Union[str, Dict]]]): The list of documents to filter.
+        timestamp (Union[str, None]): The timestamp to filter against.
+        ids (List[str]): The list of ids to filter against.
+        data_type (str): The type of data.
+    Returns: (List[Dict[str, Union[str, Dict]]]): The filtered list of documents.
+    """
+    if not documents or not timestamp or not ids:
+        return documents
 
-    demisto.debug(f'RadwareCloudDDoS: {start_time=}, {end_time=}, {skip=}')
-    response = client.get_events(start_time, end_time, skip, 700)
-    documents = response.get("documents")
-    demisto.debug(f'RadwareCloudDDoS: {len(documents)=}')
-    new_continue_fetch_events = {}
+    def get_timestamp(data: Dict[str, Union[str, Dict]]) -> Union[str, None]:
+        if data_type == 'events':
+            return data['endTimestamp']
+        else:
+            return data.get('context', {}).get('_timestamp')
 
-    if documents:
-        latest_timestamp_events = documents[0]["endTimestamp"]
-        demisto.debug(f'RadwareCloudDDoS: {latest_timestamp_events=}')
-        if len(documents) == 700:
-            demisto.debug('RadwareCloudDDoS: found next page')
-            new_continue_fetch_events = {'end_time': end_time, 'start_time': start_time, 'fetched_events': len(documents)+skip}
+    filtered_documents = []
+    for doc in documents:  # Iterate through the original list
```

Comment:
Add a condition prior to the for loop to check that the first element's timestamp is in the relevant timestamp. If not, we can finish filtering.